nailgun/entities.py

```diff
@@ -2740,17 +2870,41 @@ def read(self, entity=None, attrs=None, ignore=()):
         return super(PuppetModule, self).read(entity, attrs, ignore)
 
 
-class Realm(Entity):
+class Realm(
```

Comment:
The "robustness" branch I keep talking about provides some nicer error messages, among other things. See:

```python
>>> import robottelo; from nailgun import entities
>>> smart_proxy = entities.SmartProxy(id=1).read()
>>> realm = entities.Realm(
...     realm_proxy=smart_proxy,
...     realm_type='Active Directory',
...     name='foo3',
... ).create(create_missing=False)
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 837, in create
    return self.read(attrs=self.create_json(create_missing))
  File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 692, in read
    in _get_entity_ids(field_name, attrs)
  File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 274, in _get_entity_ids
    attrs.keys()
nailgun.entity_mixins.MissingValueError: Cannot find a value for the "organization" field. Searched for keys named ('organization_ids', 'organization', 'organizations'), but available keys are [u'realm_proxy_id', u'name', u'updated_at', u'created_at', u'hosts_count', u'id', u'hostgroups_count', u'realm_type'].
```